pynvml/tests/test_nvml.py

```diff
@@ -132,6 +133,17 @@ def test_nvmlDeviceGetPowerUsage(ngpus, handles):
         power_watts = pynvml.nvmlDeviceGetPowerUsage( handles[i] )
         assert power_watts >= 0.0
 
+# Test pynvml.nvmlDeviceGetTotalEnergyConsumption
+def test_nvmlDeviceGetTotalEnergyConsumption(ngpus, handles):
+    for i in range(ngpus):
+        energy_joules1 = pynvml.nvmlDeviceGetTotalEnergyConsumption(handles[i])
+        time.sleep(1e-3)  # idle for 1 ms
+        energy_joules2 = pynvml.nvmlDeviceGetTotalEnergyConsumption(handles[i])
+        assert energy_joules1 > 0
+        assert energy_joules2 > 0
+        # check that idle wattage (mJ consumed across 1 ms) is non-zero
+        assert energy_joules2 > energy_joules1
```

Comment:
This test actually fails most of the time on my system (it seems that the value is updated less frequetly than 1ms).  For me, the value seems to change every 10ms or so, but I am not sure how the precision varies for different systems/configurations. 

One possible approach is to allow the change to take a full second, but to check every ms (to avoid a 1+ second run time on most systems).  This can be done by sleeping/checking in a loop (and breaking as soon as the energy consumption increases).  For example:

```python
def test_nvmlDeviceGetTotalEnergyConsumption(ngpus, handles):
    for i in range(ngpus):
        energy_joules1 = pynvml.nvmlDeviceGetTotalEnergyConsumption(handles[i])
        assert energy_joules1 > 0
        is_increasing = False
        for j in range(1000):
            time.sleep(1e-3)  # idle for 1 ms
            energy_joules2 = pynvml.nvmlDeviceGetTotalEnergyConsumption(handles[i])
            # check if the idle wattage (mJ consumed across 1 ms) is non-zero
            assert energy_joules2 >= energy_joules1
            if energy_joules2 > energy_joules1:
                is_increasing = True
                break
        assert is_increasing
```

Reply:
Good thing you checked -- thanks! I'll dig a little to see if there's something principled we can do here and otherwise go with something like this.

Reply:
I wasn't able to find info on update frequency in any of the NVML reference guides or documentation. However, [a presentation by Martin Burtscher](http://on-demand.gputechconf.com/gtc/2014/presentations/S4454-accurate-energy-meas-kepler-gpus.pdf) (supported by NSF, Nvidia, TSU) uses NVML and observes power sensor sampling rates of:
>  0.266 ms to 130 ms (7.7 Hz to 3760 Hz).

Their methodology is to use a sampling rate of 15 ms. And so that's what's done now in the most recent commit, up to a limit of 150 ms (an extra 20 ms above the upper-bound they observed).